src/dune_lang/slang.ml

```diff
@@ -137,10 +137,22 @@ let catch_undefined_var ?(loc = Loc.none) value ~fallback =
 ;;
 
 let and_absorb_undefined_var ?(loc = Loc.none) blangs =
+  let blangs : blang list =
```

Comment:
This function is named after the `And_absorb_undefined_var` constructor which suggests that all it does is invoke the constructor (at least that was my intention when naming it). I think that doing additional processing to its arguments (even if it's benign as far as the value of the expression is concerned) may lead to confusion when debugging generated slang expressions. I think the `simplify` function in this module would be a more suitable place for this logic.

Reply:
If you look at `Predicate_lang`, we put similar simplifications in such smart constructors. I personally like that it makes it impossible to construct non-canonical values (at least if we make the type private). I can move this to `simplify` though

Reply:
Moved everything to simplify. I still like making the "smart" constructors a little smarter, as it's helpful to know that you're always dealing with simplified expressions, but for this is more consistent.